pkg/virtctl/vm/vm_test.go

```diff
@@ -449,15 +448,21 @@ var _ = Describe("VirtualMachine", func() {
 			table.Entry("removevolume name, missing required volume-name", "removevolume", "required flag(s)", "testvmi"),
 			table.Entry("removevolume name, invalid extra parameter", "removevolume", "unknown flag", "testvmi", "--volume-name=blah", "--invalid=test"),
 		)
-
-		It("addvolume should fail when no source is found", func() {
+		table.DescribeTable("should fail addvolume when no source is found according to option", func(isDryRun bool) {
 			kubecli.MockKubevirtClientInstance.EXPECT().CdiClient().Return(cdiClient)
 			kubecli.MockKubevirtClientInstance.EXPECT().CoreV1().Return(coreClient.CoreV1())
-			cmd := tests.NewVirtctlCommand("addvolume", "testvmi", "--volume-name=testvolume")
-			res := cmd.Execute()
+			commandAndArgs := []string{"addvolume", "testvmi", "--volume-name=testvolume"}
+			if isDryRun {
+				commandAndArgs = append(commandAndArgs, "--dry-run")
+			}
+			cmdAdd := tests.NewRepeatableVirtctlCommand(commandAndArgs...)
+			res := cmdAdd()
 			Expect(res).ToNot(BeNil())
 			Expect(res.Error()).To(ContainSubstring("Volume testvolume is not a DataVolume or PersistentVolumeClaim"))
-		})
+		},
+			table.Entry("with default", false),
```

Comment:
nit: even more generalized - instead of the `true` vs. `false` in the entries we could just put the slice with the args in here, then we would get rid of the `if` and could remove four lines.

Reply:
Thanks, @dhiller for the feedback. The first implementation was made similarly but to increase readability ([#6931 (comment)](https://github.com/kubevirt/kubevirt/pull/6931#discussion_r765676622)) I changed the code. Great to hear some other points of view for future development